utils/generate-command-code.py

```diff
@@ -94,6 +94,58 @@ def check_command_key_specs(command):
     return True
 
 
+def check_implicit_acl_categories(command):
+    """
+    Validate that ACL categories in JSON files follow rules:
+    
+    Implicit ACL category rules (from setImplicitACLCategories in server.c):
+    1. WRITE flag - WRITE category
+    2. READONLY flag (and NOT SCRIPTING category) - READ category
+    3. ADMIN flag - ADMIN and DANGEROUS categories
+    4. PUBSUB flag - PUBSUB category
+    5. FAST flag - FAST category
+    6. BLOCKING flag - BLOCKING category
+    7. If NOT FAST category - SLOW category
+    """
+    command_flags = set(command.desc.get("command_flags", []))
+    acl_categories = set(command.desc.get("acl_categories", []))
+    
+    errors = []
+    
+    if "WRITE" in command_flags and "WRITE" not in acl_categories:
+        errors.append("WRITE flag requires WRITE category")
```

Comment:
Let's also check the reverse, that not WRITE flag implies not WRITE category.

We can check both directions by checking equivalence:

```suggestion
    if ("WRITE" in command_flags) != ("WRITE" in acl_categories):
        errors.append("WRITE flag and WRITE category must match")
```